### PR TITLE
separate js error and validation error

### LINKED ISSUE #822: [Bug][High][Release] When there is a binding error we should show error instead of undefined in current value

### PR DESCRIPTION
**Pull request by ApekshaBhosale**
## Description
Show js error during evaluation and send error it to debugger

Fixes #822 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/show-js-error-rewrite 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 48.93 **(0)** | 28.67 **(-0.07)** | 26.77 **(0)** | 49.29 **(-0.02)**
 :red_circle: | app/client/src/components/editorComponents/CodeEditor/EvaluatedValuePopup.tsx | 66.23 **(0)** | 61.67 **(-4.4)** | 44 **(0)** | 69.01 **(0)**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/index.tsx | 73.68 **(0.14)** | 53.15 **(0)** | 56.67 **(0)** | 73.74 **(0.14)**
 :red_circle: | app/client/src/components/propertyControls/InputTextControl.tsx | 35.9 **(-0.94)** | 0 **(0)** | 33.33 **(0)** | 35.14 **(-0.97)**
 :red_circle: | app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx | 19.53 **(-0.8)** | 0 **(0)** | 0 **(0)** | 20.49 **(-1.25)**
 :red_circle: | app/client/src/reducers/uiReducers/debuggerReducer.ts | 37.04 **(0)** | 5.56 **(-1.58)** | 16.67 **(0)** | 40 **(0)**
 :red_circle: | app/client/src/sagas/ActionExecutionSagas.ts | 12.31 **(-0.06)** | 1.35 **(-0.05)** | 7.41 **(0)** | 14.11 **(-0.09)**
 :red_circle: | app/client/src/sagas/DebuggerSagas.ts | 15.84 **(-0.65)** | 3.28 **(-0.23)** | 28.57 **(0)** | 17.2 **(-0.98)**
 :red_circle: | app/client/src/sagas/EvaluationsSaga.ts | 20.57 **(-0.14)** | 1.67 **(-0.25)** | 5 **(0)** | 23.39 **(-0.19)**
 :red_circle: | app/client/src/selectors/propertyPaneSelectors.tsx | 60.71 **(-2.25)** | 46.15 **(0)** | 100 **(0)** | 59.26 **(-2.28)**
 :red_circle: | app/client/src/workers/DataTreeEvaluator.ts | 88.59 **(0.13)** | 77.25 **(-1.76)** | 87.14 **(0)** | 88.69 **(0.03)**</details>

### CODE REVIEW

File: app/client/src/components/propertyControls/InputTextControl.tsx [hunk 1]
```diff
@@ -51,6 +53,8 @@ export function InputText(props: {
         meta={{
           error: isValid ? "" : errorMessage,
           touched: true,
+          isJsError: jsErrorMessage && jsErrorMessage.length > 0 ? true : false,
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/InputTextControl.tsx:56]:
```suggestion
          isJsError: jsErrorMessage && jsErrorMessage.length,
```

File: app/client/src/workers/DataTreeEvaluator.ts [hunk 2]
```diff
@@ -580,17 +585,38 @@ export default class DataTreeEvaluator {
...
-          binding: js,
-        },
-      });
+      if (fullPropertyPath) {
+        const { propertyPath, entityName } = getEntityNameAndPropertyPath(
+          fullPropertyPath,
+        );
+        _.set(data, `${entityName}.jsErrorMessages.${propertyPath}`, e.message);
+        const entity = data[entityName];
+        if (isWidget(entity)) {
```
Comment (hetunandu) [inline app/client/src/workers/DataTreeEvaluator.ts:605]: Should happen for actions as well

File: app/client/src/components/propertyControls/InputTextControl.tsx (hunk 1 above)
Comment (hetunandu) [reply app/client/src/components/propertyControls/InputTextControl.tsx:56]: We can also avoid this check here completely and just pass the message. Do this check in the evaluated value popup

**Review by hetunandu — CHANGES_REQUESTED** [review]
Minor changes

File: app/client/src/workers/DataTreeEvaluator.ts (hunk 2 above)
Comment (ApekshaBhosale) [reply app/client/src/workers/DataTreeEvaluator.ts:605]: Action has taken care in actionssaga @hetunandu i am adding action error to debugger on page load only

**Review by rishabhsaxena — APPROVED** [review]
LGTM

**Review by rishabhsaxena — DISMISSED** [review]

**Review by rishabhsaxena — APPROVED** [review]

**Pull request closed without merging**